File: fizz-buzz.py
print("Fizz-Buzz by kffiatek")

game_size = 100
factors_of_interest = {3 : "Fizz", 5 : "Buzz",  6: "szakalaka"}

for i in range(1,game_size+1):  # gra zaczyna się od 1
  result=""
  player=i % 2  # gracz parzysty albo nieparzysty 0 albo 1
  for factor in factors_of_interest: # gra dla wszystkich fiz-buzów w zbiorze (oryginalnie 3 i 5)
    if (i % factor == 0):  #jeśli dzieli się przez dany element
      if result: result+="-"  #sprawdzamy, czy to pierwszy, żeby dodać myślnik, jeśli to drugi albo kolejny
      result+=factors_of_interest[factor] #dodajemy fizz buzz albo coś innego
  if not result: result = str(i) # jeśli przez nic się nie dzieli, wyświetlamy listę
  result = abs((player-1))*10*(" ") + result #gracz parzysty zostaje przesunięty w prawo o 10
  print(result)

File: fizz-buzz.cpp
#include <iostream>
#include <string>

int game_size = 100;

const char* words_t[]= {"Fizz", "Buzz", "szakalaka"};
const int values_t[] = {3, 5, 6};

int main(){
  for (int i = 1; i <= game_size; i++){
    int player = i % 2;
    int values_count =sizeof(values_t)/sizeof(values_t[0]);
    bool is_divisible = false;
    for (int j = 0; j<values_count; j++){
      if (i % values_t[j] == 0) {
        if (is_divisible){
          std::cout << "-"; 
        }
        is_divisible = true;
        std::cout << words_t[j];
      }
    }
    if (!is_divisible){
      std::cout << i;
    }
    std::cout << "\n";
  }
}